tests/src/test/java/org/eclipse/hono/tests/IntegrationTestSupport.java

```diff
@@ -841,9 +903,19 @@ public void deleteObjects(final VertxTestContext ctx) {
     public Future<?> disconnect() {
 
         final Promise<Void> legacyClientResult = Promise.promise();
-        applicationClientFactory.disconnect(legacyClientResult);
+        if (applicationClientFactory != null) {
+            applicationClientFactory.disconnect(legacyClientResult);
+        } else {
+            legacyClientResult.complete();
+        }
+
         final Promise<Void> newClientResult = Promise.promise();
-        amqpApplicationClient.disconnect(newClientResult);
+        if (applicationClient instanceof AmqpApplicationClient) {
+            ((AmqpApplicationClient) applicationClient).disconnect(newClientResult);
+        } else {
+            newClientResult.complete();
+        }
```

Comment:
looks like `ApplicationClient` is lacking a generic *close()* method to release underlying resources, doesn't it?

Reply:
Hmm. I was about to start implementing this method, but then I noticed that `ApplicationClient` actually implements `Lifecycle#stop`.

The `ProtonBasedApplicationClient` calls `HonoConnection#shutdown` in its `Lifecycle#stop` method. Both, the `shutdown` and the `disconnect` method end up calling `HonoConnectionImpl#closeConnection`.

So I think calling `Lifecycle#stop` actually is already the generic _close()_ method?